src/test/integration/IntegrationDeployer.t.sol

```diff
@@ -131,503 +94,445 @@ abstract contract IntegrationDeployer is ExistingDeploymentParser {
+        config.governance.pauserRegistry = new PauserRegistry(PAUSER.toArray(), UNPAUSER);
+        config.governance.executorMultisig = address(proxyAdmin().owner());
 
         _deployProxies();
-        _deployImplementations();
         _upgradeProxies();
         _initializeProxies();
 
         // Place native ETH first in `allStrats`
-        // This ensures when we select a nonzero number of strategies from this array, we always
-        // have beacon chain ETH
-        ethStrats.push(BEACONCHAIN_ETH_STRAT);
+        // This ensures when we select a nonzero number of strategies from this array, we always have beacon chain ETH.
         allStrats.push(BEACONCHAIN_ETH_STRAT);
         allTokens.push(NATIVE_ETH);
 
-        // Deploy and configure strategies and tokens
+        // Deploy and configure strategies and tokens, deploy half of them using the strategy factory.
         for (uint i = 1; i < NUM_LST_STRATS + 1; ++i) {
             string memory name = string.concat("LST-Strat", cheats.toString(i), " token");
             string memory symbol = string.concat("lstStrat", cheats.toString(i));
-            // Deploy half of the strategies using the factory.
             _newStrategyAndToken(name, symbol, 10e50, address(this), i % 2 == 0);
         }
 
+        // Whitelist the strategies
+        cheats.prank(strategyManager().strategyWhitelister());
+        strategyManager().addStrategiesToDepositWhitelist(allStrats);
+
         maxUniqueAssetsHeld = allStrats.length;
 
         // Create time machine and beacon chain. Set block time to beacon chain genesis time and starting block number
         BEACON_GENESIS_TIME = GENESIS_TIME_LOCAL;
         cheats.warp(BEACON_GENESIS_TIME);
         cheats.roll(10_000);
-        timeMachine = new TimeMachine();
-        beaconChain = new BeaconChainMock(eigenPodManager, BEACON_GENESIS_TIME);
-
+        _deployTimeMachineAndBeaconChain();
         // Set the `pectraForkTimestamp` on the EigenPodManager. Use pectra state
-        cheats.startPrank(executorMultisig);
-        eigenPodManager.setProofTimestampSetter(executorMultisig);
-        eigenPodManager.setPectraForkTimestamp(BEACON_GENESIS_TIME);
+        cheats.startPrank(executorMultisig());
+        eigenPodManager().setProofTimestampSetter(executorMultisig());
+        eigenPodManager().setPectraForkTimestamp(BEACON_GENESIS_TIME);
         cheats.stopPrank();
     }
 
-    /// Parse existing contracts from mainnet
-    function _setUpMainnet() public virtual noTracing {
-        console.log("Setting up `%s` integration tests:", "MAINNET_FORK".green().bold());
-        console.log("RPC:", cheats.rpcUrl("mainnet"));
-        console.log("Block:", mainnetForkBlock);
+    function _setUpFork() public virtual {
+        _setUpFork(true);
+    }
 
-        cheats.createSelectFork(cheats.rpcUrl("mainnet"), mainnetForkBlock);
+    /// @dev Sets up the integration tests for mainnet.
+    function _setUpFork(bool upgrade) public virtual {
```

Comment:
This function is pretty messy, can we clean it up?

- Confusion between `bool upgrade` vs `forkConfig.upgradeBeforeTesting` vs `isUpgraded`
- `forkConfig.supportEigenPodTests` evaluated in multiple places; let's please keep that to 1 (im not sure what this is for in the first place though)
- First part of the method does either `_executeTimelockUpgrade` or `_deployProxies + _upgradeProxies`. Later in the method, we might also do `_upgradeMainnetContracts`. Is that mutually exclusive, or can that happen in addition to the earlier upgrade stuff? Logic should be more clear.

Reply:
Oh, I realized `_upgradeMainnetContracts` is just `_upgradeProxies`, but disguised... let's collapse these definitions and condense upgrade logic pls